glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt

```diff
@@ -53,4 +103,361 @@ class GleanTest {
+            StringsStorageEngine.getSnapshot(storeName = "store1", clearStore = false)
+        )
+
+        Glean.initialized = true
+        */
+    }
+
+    @Test
+    fun `Initializing twice is a no-op`() {
+        val beforeConfig = Glean.configuration
+
+        Glean.initialize(ApplicationProvider.getApplicationContext())
+
+        val afterConfig = Glean.configuration
+
+        assertSame(beforeConfig, afterConfig)
+    }
+
+    @Test
+    fun `Don't handle events when uninitialized`() {
+        val gleanSpy = spy<GleanInternalAPI>(GleanInternalAPI::class.java)
+
+        //gleanSpy.initialized = false
+        gleanSpy.testDestroyGleanHandle()
+        runBlocking {
+            gleanSpy.handleBackgroundEvent()
+        }
+        assertFalse(isWorkScheduled(PingUploadWorker.PING_WORKER_TAG))
+    }
+
+    @Test
+    fun `Don't schedule pings if metrics disabled`() {
+        Glean.setUploadEnabled(false)
+
+        runBlocking {
+            Glean.handleBackgroundEvent()
+        }
+        assertFalse(isWorkScheduled(PingUploadWorker.PING_WORKER_TAG))
+    }
+
+    @Test
+    fun `Don't schedule pings if there is no ping content`() {
+        resetGlean(getContextWithMockedInfo())
+
+        runBlocking {
+            Glean.handleBackgroundEvent()
+        }
+
+        // We should only have a baseline ping and no events or metrics pings since nothing was
+        // recorded
+        val files = File(Glean.getDataDir(), PingUploader.PINGS_DIR).listFiles()
+
+        // Make sure only the baseline ping is present and no events or metrics pings
+        assertEquals(1, files.count())
+        val file = files.first()
+        BufferedReader(FileReader(file)).use {
+            val lines = it.readLines()
+            assert(lines[0].contains("baseline"))
+        }
+    }
+
+    @Test
+    fun `The appChannel must be correctly set, if requested`() {
+        // No appChannel must be set if nothing was provided through the config
+        // options.
+        resetGlean(getContextWithMockedInfo(), Configuration())
+        assertFalse(GleanInternalMetrics.appChannel.testHasValue())
+
+        // The appChannel must be correctly reported if a channel value
+        // was provided.
+        val testChannelName = "my-test-channel"
+        resetGlean(getContextWithMockedInfo(), Configuration(channel = testChannelName))
+        assertTrue(GleanInternalMetrics.appChannel.testHasValue())
+        assertEquals(testChannelName, GleanInternalMetrics.appChannel.testGetValue())
+    }
+
+    // glean-ac test removed.
+    // `client_id and first_run_date metrics should be copied from the old location` was here.
+    // 1539480 BACKWARD COMPATIBILITY HACK that is not needed anymore.
+
+    // glean-ac test removed.
+    // `client_id and first_run_date metrics should not override new location` was here.
+    // 1539480 BACKWARD COMPATIBILITY HACK that is not needed anymore.
+
+    @Ignore("This should probably be implemented in Rust")
```

Comment:
Uh, this is important. Do we have a bug already?

Reply:
Right, filed [bug 1554082](https://bugzilla.mozilla.org/show_bug.cgi?id=1554082) for it.